### PR TITLE
feat: [IAC-2870]: iacm plan command

### PR DESCRIPTION
**Pull request by dwarren-dev**

### CODE REVIEW

File: account/account.go [hunk 1]
```diff
@@ -2,11 +2,12 @@ package account
 
 import (
 	"fmt"
-	"github.com/urfave/cli/v2"
```
Comment (scottyw-harness) [inline account/account.go:5]: Maybe revert this?

File: iacm-plan.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+		return err
+	}
+
+	warning, err := validateWorkdingDirectory(wd, ws)
+	if err != nil {
+		fmt.Println(utils.GetColoredText(err.Error(), color.FgRed))
+		return err
+	}
+
+	fmt.Println(warning)
```
Comment (scottyw-harness) [inline iacm-plan.go:75]: Debug?

File: main.go [hunk 3]
```diff
@@ -580,31 +582,73 @@ func main() {
...
+							altsrc.NewStringFlag(&cli.StringFlag{
+								Name:     "project-id",
+								Usage:    "provide a Project Identifier",
+								Required: true,
+							}),
+						},
+						Action: func(context *cli.Context) error {
+							iacmClient := client.NewIacmClient(
+								shared.CliCdRequestData.Account,
+								fmt.Sprintf("%sgateway", shared.CliCdRequestData.BaseUrl),
```
Comment (scottyw-harness) [inline main.go:644]: Any danger of slashes messing this up e.g. if BaseURL is missing the slash cos a user specified it?

**Review by scottyw-harness — APPROVED** [review]
Looks grand. You deleted a binary called "harnes" which seems fine too.

File: iacm-plan.go (hunk 2 above)
Comment (dwarren-dev) [reply iacm-plan.go:75]: no that gives the warning of which part of the fs the upload happens from, that warning messages is determined by validateWorkingDirectory

File: main.go (hunk 3 above)
Comment (dwarren-dev) [reply main.go:644]: yea we'll need to harden this eventually

**Pull request merged**